src/logic/AppCenterCrashes.swift

```diff
@@ -4,41 +4,26 @@ import AppCenterCrashes
 
 class AppCenterCrash: NSObject, CrashesDelegate {
     static let secret = Bundle.main.object(forInfoDictionaryKey: "AppCenterSecret") as! String
-
-    override init() {
-        super.init()
+    // we need to call this again if the setting changes so make it a function
+    func setDefaultNetworkRequestsPolicy(override: Bool = false) {
+        // this will default to false thus not sending anything until opted-in
+        // override is to override temporarily on a case-by-case basis
+        AppCenter.networkRequestsAllowed = override || Preferences.crashPolicy == .always
+    }
```

Comment:
We may need to call this twice so made this a function. If the user's preference is to "ask", this will restrict network communications by default, so we need to call it again with the override flag when the user clicks "send" in the modal following a crash to temporarily allow network requests.